viam-cartographer/src/main.cc

```diff
@@ -53,7 +53,14 @@ int main(int argc, char** argv) {
     // This log line is needed by rdk to get the port.
     LOG(INFO) << "Server listening on " << *selected_port << "\n";
 
-    slamService.RunSLAM();
+    try {
+        slamService.RunSLAM();
```

Comment:
Note: This is being added so that the program terminates in all cases if https://github.com/viamrobotics/viam-cartographer/pull/25/files#diff-92ff8a5cd3a57430d78c6ba3b61a91d1810399b77613eece3c55841868d2822eR384 throws.

Reply:
*from the main thread